### PR TITLE
Oauth secrets

### PR DESCRIPTION
**Pull request by A3a3e1**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,4 +150,33 @@ If not specified, the message will be soft-deleted (moved to the folder with pro
...
+  "toRecipients": [
+    {
+      "emailAddress": {
+        "address": "email@example.com",
+        "name": "John"
+      }
+    }
+  ],
+  "body": {
+    "content": "sdf",
```
Comment (A3a3e1) [inline README.md:176]:
```suggestion
    "content": "Hello, I am an email content text",
```

File: README.md [hunk 2]
```diff
@@ -31,30 +31,51 @@
...
+* Contacts.Read
+* Profile
+* Calendars.ReadWrite
+* Mail.ReadWrite
+* Mail.Send
+
+offline_access is required for each credential.
+Example of the scopes for the `Send Mail` action: `offline_access Mail.Send`
+
+**Please Note!** To verify be able to verify the credentials you need these scopes: `offline_access User.Read`
```
Comment (if0s) [inline README.md:71]:
```suggestion
**Please Note!** To be able to verify the credentials you need these scopes: `offline_access User.Read`
```

**Review by if0s — APPROVED** [review]

File: lib/client.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  async getNewAccessToken() {
+    let fields;
+    if (this.cfg.secretId) {
+      this.logger.debug('Fetching credentials by secretId');
+      const response = await getSecret.call(this, this.cfg.secretId);
+      this.accessToken = response.credentials.access_token;
+      fields = response.credentials.fields;
+      this.logger.debug(`${JSON.stringify(response, null, 2)}`);
```
[the remark this answers is not in the document]
Comment (A3a3e1) [reply lib/client.js:81]: Of course it's not. Left from the debug :) Will remove

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,6 @@
+## 2.0.0 (July 19, 2023)
+* Breaking! Reworked authentication mechanism - implemented Secrets feature
```
Comment (A3a3e1) [inline CHANGELOG.md:2]:
```suggestion
* Breaking change! Reworked authentication mechanism - implemented Secrets feature
* Add new action - Send Mail
```

**Pull request merged**